apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/DocumentLogs/DocumentLogInfo/index.tsx

```diff
@@ -0,0 +1,68 @@
+'use client'
+
+import { useCallback, useEffect, useState } from 'react'
+
+import type { DocumentLogWithMetadata } from '@latitude-data/core'
+import { ProviderLog } from '@latitude-data/core/browser'
+import { TabSelector } from '@latitude-data/web-ui'
+import { getProviderLogsForDocumentLogAction } from '$/actions/providerLogs/getProviderLogsForDocumentLogAction'
+import { useServerAction } from 'zsa-react'
+
+import { DocumentLogMessages } from './Messages'
+import { DocumentLogMetadata } from './Metadata'
+
+export function DocumentLogInfo({
+  documentLog,
+}: {
+  documentLog: DocumentLogWithMetadata
+}) {
+  const [selectedTab, setSelectedTab] = useState<string>('metadata')
+  const [providerLogs, setProviderLogs] = useState<ProviderLog[]>()
+  const { execute: executeFetchProviderLogsAction } = useServerAction(
+    getProviderLogsForDocumentLogAction,
+  )
+
+  const fetchProviderLogs = useCallback(
+    async (documentLogId: number) => {
+      setProviderLogs(undefined)
+      const [logs, error] = await executeFetchProviderLogsAction({
```

Comment:
For another PR but this could have a SWR store to be cached and avoid refetching everytime

Reply:
Yeah, much better!